u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -192,6 +194,85 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     }
   }
 
+  public static TransferAccessResponse decodeTransferAccessResponse(byte[] data)
+      throws U2FException {
+    int sequenceNumberSize = 1;
+    int publicKeySize = 65;
+    int appIdSize = 32;
+
+    try {
+      DataInputStream inputStream = new DataInputStream(new ByteArrayInputStream(data));
+      byte controlByte = inputStream.readByte();
+
+      if (!inputStream.markSupported()) {
+        throw new U2FException("Stream doesn't support mark and reset.");
+      }
+      inputStream.mark(inputStream.available());
+      int numberOfTransferAccessMessages = inputStream.readUnsignedByte();
+      inputStream.reset();
+
+      TransferAccessMessage[] transferAccessMessages =
+          new TransferAccessMessage[numberOfTransferAccessMessages];
+
+      for (int i = 0; i < numberOfTransferAccessMessages; i++) {
```

Comment:
Done, but a potential issue may need to be discussed:

If I pass a [DataInputStream](https://docs.oracle.com/javase/7/docs/api/java/io/DataInputStream.html), I don't know that there is a way to check the underlying stream to make sure it is at least the type we expect (ByteArrayInputStream). I can, however, just treat it as a byte array since this should be the only function calling it for now.